packages/auth0/data_stream/logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,1110 @@
+          type:
+            - info
+          category:
+            - intrusion_detection
+          action: login-with-breached-password
+        scph:
+          classification: "Other"
+          value: "Success Post Change Password Hook"
+          type:
+            - info
+          action: success-post-change-password-hook
+        sd:
+          classification: "Other"
+          value: "Success delegation"
+          type:
+            - info
+          action: success-delegation
+        si:
+          classification: "Other"
+          value: "Successfully accepted a user invitation"
+          type:
+            - info
+          action: successfully-accepted-user-invitation
+        ss:
+          classification: "Other"
+          value: "Success Signup"
+          type:
+            - info
+          action: success-signup
+    source: |-
+         def eventType = ctx.auth0.logs.data.type;
+         def actions = params.get('actions');
+         def actionData = actions.get(eventType);
+         if (actionData == null) {
+             ctx.event.action = 'unknown-' + eventType;
+             ctx.event.type = ['info'];
+             return;
+         }
+         // overwrite type abbreviation with actual value
+         def eventTypeVal = actionData.get('value');
+         if (eventTypeVal != null) {
+             ctx.auth0.logs.data.type = eventTypeVal;
+         }
+         // event.type
+         def actionType = actionData.get('type');
+         if (actionType != null) {
+           ctx.event.type = new ArrayList(actionType);
+         }
+         // event.category
+         def actionCategory = actionData.get('category');
+         if (actionCategory != null) {
+           for (def c : actionCategory) {
+             ctx.event.category.add(c);
+           }
+         }
+         // event.action
+         def action = actionData.get('action');
+         if (action != null) {
+           ctx.event.action = action;
+         }
+         // auth0 event category / classification group
+         def classification = actionData.get('classification');
+         if (classification != null) {
+           ctx.auth0.logs.data.classification = classification;
+         }
+         // event.outcome
+         if (classification.toLowerCase().contains("success")) {
+           ctx.event.outcome = "success";
+         } else if (classification.toLowerCase().contains("failure")) {
+           ctx.event.outcome = "failure";
+         } else {
+           ctx.event.outcome = "unknown";
+         }
+- date:
+    if: "ctx.auth0.logs.data.type == 'Successful login' && ctx.auth0.logs.data.details.initiatedAt != null"
+    field: auth0.logs.data.details.initiatedAt
+    target_field: auth0.logs.data.login.initiatedAt
+    formats:
+      - UNIX_MS
+- date:
+    if: "ctx.auth0.logs.data.type == 'Successful login' && ctx.auth0.logs.data.details.completedAt != null"
```

It looks like when `completedAt` exists it is always a UNIX_MS timestamp, so can we drop the check for `ctx.auth0.logs.data.type`?

```suggestion
    if: "ctx?.auth0?.logs?.data?.details?.completedAt != null"
```

I have a similar question regarding the surrounding processing that are checking the data type. Do these fields require conditional processing based on the event type? It seems like when the field exists it always requires the same type of processing. My preference is towards fewer invariants in pipelines to make them simpler to follow.